sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -98,7 +99,23 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
     programOffering.creator = { id: userId } as User;
     // When creating a new offering, parent id and the primary id are the same.
     programOffering.parentOffering = programOffering;
-    return this.repo.save(programOffering);
+
+    try {
+      return await this.repo.save(programOffering);
+    } catch (error: unknown) {
+      if (error instanceof QueryFailedError) {
```

Comment:
These checks are now happening in many places. As a suggestion, if we change `PostgresDriverError` to a class, can the code be simplified as the below?
```ts
if (error instanceof PostgresDriverError && error.constraint ===
  DatabaseConstraintNames.LocationIDProgramIDOfferingNameStudyDatesYearOfStudyIndex) {
    throw new CustomNamedError(
      "Duplication error. An offering with the same name, year of study, start date and end date was found.",
      OFFERING_SAVE_UNIQUE_ERROR,
    );
  }
}
```